packages/flutter/lib/src/material/slider_theme.dart

```diff
@@ -489,13 +543,94 @@ class SliderThemeData extends Diagnosticable {
     properties.add(DiagnosticsProperty<SliderComponentShape>('valueIndicatorShape', valueIndicatorShape, defaultValue: defaultData.valueIndicatorShape, level: DiagnosticLevel.debug));
     properties.add(EnumProperty<ShowValueIndicator>('showValueIndicator', showValueIndicator, defaultValue: defaultData.showValueIndicator));
     properties.add(DiagnosticsProperty<TextStyle>('valueIndicatorTextStyle', valueIndicatorTextStyle, defaultValue: defaultData.valueIndicatorTextStyle));
   }
 }
 
+/// Base class for slider track shapes.
+///
+/// Create a subclass of this if you would like a custom track shape.
+///
+/// See also:
+///
+///  * [RectangularSliderTrackShape] for the default track shape.
+///  * [SliderTickShape] for the base class for custom a tick shape.
+///  * [SliderComponentShape] for the base class for custom a component shape.
+abstract class SliderTrackShape {
+  /// Abstract const constructor. This constructor enables subclasses to provide
+  /// const constructors so that they can be used in const expressions.
+  const SliderTrackShape();
+
+  /// Returns the preferred bounds of the shape, based on the given conditions.
+  Rect getPreferredRect({
+    RenderBox parentBox,
+    Offset offset = Offset.zero,
+    SliderThemeData sliderTheme,
+    bool isEnabled,
+    bool isDiscrete,
+  });
+
+  /// Called to paint a track.
+  void paint(PaintingContext context, Offset offset, {
+    RenderBox parentBox,
+    SliderThemeData sliderTheme,
+    Animation<double> enableAnimation,
+    List<double> trackDividers,
+    List<bool> trackSegmentsActive,
+    bool isDiscrete,
+    bool isEnabled,
+  });
+}
+
+/// Base class for slider tick shapes.
+///
+/// Create a subclass of this if you would like a custom slider tick shape.
+/// This is a simplified version of [SliderComponentShape] with a
+/// [SliderThemeData] passed when getting the preferred size.
+///
+/// See also:
+///
+///  * [RoundSliderTickShape] for a simple example of a tick shape.
+///  * [SliderTrackShape] for the base class for custom a track shape.
+///  * [SliderComponentShape] for the base class for custom a component shape.
+abstract class SliderTickShape {
+  /// Abstract const constructor. This constructor enables subclasses to provide
+  /// const constructors so that they can be used in const expressions.
+  const SliderTickShape();
+
+  /// Returns the preferred size of the shape, based on the given conditions.
+  Size getPreferredSize({
+    bool isEnabled,
+    SliderThemeData sliderTheme,
+  });
+
+  /// Paints the shape, taking into account the state passed to it.
+  ///
+  /// [center] Is the center offset of the tick to be painted.
+  ///
+  /// [activationAnimation] is an animation triggered when the user beings
+  /// to interact with the slider. It reverses when the user stops interacting
+  /// with the slider.
+  ///
+  /// [enableAnimation] is an animation triggered when the [Slider] is enabled,
```

Comment:
What does it mean for the slider to be enabled? Good to link to that here.

Reply:
I linked to Slider.isInteractive, but am now realizing that [Slider] is not clickable within the IDE. But I don't think I can import it because Slider depends on SliderTheme. Is there a way to link an outside class like this?

Reply:
[Slider] should work (eventually), best to leave the link as it is.

Reply:
Done